internal/configuration/configuration.go

```diff
@@ -131,23 +132,37 @@ func (m *Manager) Update(message models.DeviceConfigurationMessage) error {
 		return nil
 	}
 
-	log.Tracef("updating configuration: %v", message)
 	var errors error
+
+	newJson, err := json.Marshal(message)
+	if err != nil {
+		errors = multierror.Append(errors, fmt.Errorf("cannot marshal new message to JSON: %s", err))
+		return errors
+	}
+
+	oldJson, err := json.Marshal(m.deviceConfiguration)
+	if err != nil {
+		errors = multierror.Append(errors, fmt.Errorf("cannot marshal old configuration to JSON: %s", err))
+		return errors
+	}
+
+	log.Infof("updating configuration. New config: %s\nOld config: %s", newJson, oldJson)
 	for _, observer := range m.observers {
 		err := observer.Update(message)
 		if err != nil {
 			errors = multierror.Append(errors, fmt.Errorf("running config observer failed: %s", err))
 		}
 	}
 
-	file, err := json.MarshalIndent(message, "", " ")
+	file := bytes.Buffer{}
+	err = json.Indent(&file, newJson, "", " ")
 	if err != nil {
-		errors = multierror.Append(errors, fmt.Errorf("cannot unmarshal message JSON: %s", err))
+		errors = multierror.Append(errors, fmt.Errorf("cannot marshal message to JSON: %s", err))
```

Comment:
not biggy, but the message should say 'cannot indent message to pretty JSON format'